javatests/arcs/core/data/ParticleSpecTest.kt

```diff
@@ -48,6 +69,65 @@ class ParticleSpecTest {
     assertFailsWith<IllegalArgumentException> { spec.dataflowType }
   }
 
+  @Test
+  fun egressType_noEgressType_returnsNull() {
+    val spec = createSpec()
+
+    assertThat(spec.egressType).isNull()
+  }
+
+  @Test
+  fun egressType_typeNotStated_returnsNull() {
+    val spec = createSpec(annotations = listOf(Annotation.createEgress()))
+
+    assertThat(spec.egressType).isNull()
+  }
+
+  @Test
+  fun egressType_typeStated_returnsTypeName() {
+    val spec = createSpec(annotations = listOf(Annotation.createEgress("MyEgressType")))
+
+    assertThat(spec.egressType).isEqualTo("MyEgressType")
+  }
+
+  @Test
+  fun egressType_dataflowTypeIsIngress_returnsNull() {
+    val spec = createSpec(annotations = listOf(Annotation.ingress))
+
+    assertThat(spec.egressType).isNull()
+  }
+
+  @Test
+  fun egressType_dataflowTypeIsIsolated_returnsNull() {
+    val spec = createSpec(annotations = listOf(Annotation.isolated))
+
+    assertThat(spec.egressType).isNull()
+  }
+
+  @Test
+  fun egressType_dataflowTypeIsIngressAndEgressWithoutArgument_returnsNull() {
+    val spec = createSpec(
+      annotations = listOf(
+        Annotation.createEgress(),
+        Annotation.ingress
+      )
+    )
+
+    assertThat(spec.egressType).isNull()
+  }
+
+  @Test
+  fun egressType_dataflowTypeIsIngressAndEgressWithArgument_returnsTypeName() {
```

Comment:
do we need tests for cases when ingressType is specified?

Reply:
👍